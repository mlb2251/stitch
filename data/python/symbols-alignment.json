[
  "(+ 1 &x1 2 &x2 3 &x3 4 &x4 5 &x5 6 &x6 7 &x7 8 &x8 9 &x9 10 &x10)",
  "(+ 1 &y1 2 &y2 3 &y3 4 &y4 5 &y5 6 &y6 7 &y7 8 &y8 9 &y9 10 &y10)"
]