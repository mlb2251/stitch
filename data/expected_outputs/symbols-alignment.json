{
  "cmd": "/home/kavi/mit/stitch/target/debug/deps/integration_tests-21175b69aaefd7be",
  "args": {
    "iterations": 2,
    "abstraction_prefix": "fn_",
    "previous_abstractions": 0,
    "shuffle": false,
    "truncate": null,
    "no_opt": false,
    "silent": false,
    "verbose_rewrite": false,
    "step": {
      "max_arity": 3,
      "threads": 1,
      "no_stats": false,
      "batch": 1,
      "dynamic_batch": false,
      "eta_long": false,
      "no_curried_metavars": false,
      "no_curried_bodies": false,
      "inv_candidates": 1,
      "hole_choice": "DepthFirst",
      "cost": {
        "cost_lam": 1,
        "cost_app": 1,
        "cost_var": 100,
        "cost_ivar": 100,
        "cost_prim_default": 100,
        "cost_prim": "{}"
      },
      "no_mismatch_check": false,
      "follow": null,
      "follow_types": null,
      "follow_prune": false,
      "verbose_worklist": false,
      "verbose_best": false,
      "print_stats": 0,
      "show_rewritten": false,
      "rewritten_dreamcoder": false,
      "rewritten_intermediates": false,
      "inv_arg_cap": false,
      "allow_single_task": false,
      "no_opt_single_use": false,
      "no_opt_upper_bound": false,
      "no_opt_force_multiuse": false,
      "no_opt_useless_abstract": false,
      "no_opt_arity_zero": false,
      "no_other_util": false,
      "structure_penalty": 1.0,
      "rewrite_check": false,
      "utility_by_rewrite": false,
      "dreamcoder_comparison": false,
      "quiet": false,
      "fused_lambda_tags": {
        "tags": null
      },
      "tdfa": {
        "tdfa_json_path": null,
        "tdfa_root": null,
        "valid_metavars": null,
        "valid_roots": null,
        "tdfa_non_eta_long_states": null,
        "tdfa_split": null
      },
      "symvar": {
        "symvar_prefix": "&"
      }
    }
  },
  "original_cost": 4240,
  "final_cost": 2220,
  "compression_ratio": 1.90990990990991,
  "num_abstractions": 1,
  "original": [
    "(+ 1 &x1 2 &x2 3 &x3 4 &x4 5 &x5 6 &x6 7 &x7 8 &x8 9 &x9 10 &x10)",
    "(+ 1 &y1 2 &y2 3 &y3 4 &y4 5 &y5 6 &y6 7 &y7 8 &y8 9 &y9 10 &y10)"
  ],
  "rewritten": [
    "(fn_0 &x9 &x8 &x7 &x6 &x5 &x4 &x3 &x2 &x1 &x10)",
    "(fn_0 &y9 &y8 &y7 &y6 &y5 &y4 &y3 &y2 &y1 &y10)"
  ],
  "rewritten_dreamcoder": null,
  "abstractions": [
    {
      "body": "(+ 1 #8 2 #7 3 #6 4 #5 5 #4 6 #3 7 #2 8 #1 9 #0 10)",
      "dreamcoder": "#(lambda (lambda (lambda (lambda (lambda (lambda (lambda (lambda (lambda (+ 1 $0 2 $1 3 $2 4 $3 5 $4 6 $5 7 $6 8 $7 9 $8 10))))))))))",
      "arity": 9,
      "name": "fn_0",
      "utility": 901,
      "final_cost": 2220,
      "compression_ratio": 1.90990990990991,
      "cumulative_compression_ratio": 1.90990990990991,
      "num_uses": 2,
      "rewritten": null,
      "rewritten_dreamcoder": null,
      "uses": [
        {
          "fn_0 &x9 &x8 &x7 &x6 &x5 &x4 &x3 &x2 &x1": "(+ 1 &x1 2 &x2 3 &x3 4 &x4 5 &x5 6 &x6 7 &x7 8 &x8 9 &x9 10)"
        },
        {
          "fn_0 &y9 &y8 &y7 &y6 &y5 &y4 &y3 &y2 &y1": "(+ 1 &y1 2 &y2 3 &y3 4 &y4 5 &y5 6 &y6 7 &y7 8 &y8 9 &y9 10)"
        }
      ],
      "dc_comparison_millis": null,
      "tdfa_annotation": null,
      "variable_types": [
        "M",
        "S",
        "S",
        "M",
        "S",
        "S",
        "S",
        "M",
        "S"
      ]
    }
  ]
}